### PR TITLE
refactor: change address from bytes to bech32 string

### PR DESCRIPTION
**Pull request by akhilkumarpilli**

### CODE REVIEW

File: integration/e2e_test.go [hunk 1]
```diff
@@ -78,7 +78,7 @@ func (s *IntegrationTestSuite) TearDownSuite() {
 	_, err = deploycli.TxCloseDeploymentExec(
 		val.ClientCtx,
 		keyTenant.GetAddress(),
-		fmt.Sprintf("--owner=%s", createdDep.Groups[0].GroupID.Owner.String()),
+		fmt.Sprintf("--owner=%s", createdDep.Groups[0].GroupID.Owner),
```
Comment (boz) [inline integration/e2e_test.go:81]: :+1:

File: sdkutil/object_id.go [hunk 2]
```diff
@@ -104,21 +107,25 @@ func ReflectBlockID(id interface{}) BlockID {
 				}
 			}
 		case "Provider":
-			if field.Type() == reflect.TypeOf(sdk.AccAddress{}) {
-				x, ok := field.Interface().(sdk.AccAddress)
-				if ok && len(x) == sdk.AddrLen {
-					b.Provider = &x
+			if field.Type() == reflect.TypeOf(string("")) {
```
Comment (boz) [inline sdkutil/object_id.go:110]: I think we'll have to take a look to see if this code makes sense anymore.

File: x/deployment/types/msgs.go [hunk 3]
```diff
@@ -37,7 +37,12 @@ func (msg MsgCreateDeployment) GetSignBytes() []byte {
 
 // GetSigners defines whose signature is required
 func (msg MsgCreateDeployment) GetSigners() []sdk.AccAddress {
-	return []sdk.AccAddress{msg.ID.Owner}
+	owner, err := sdk.AccAddressFromBech32(msg.ID.Owner)
```
Comment (boz) [inline x/deployment/types/msgs.go:40]: Can you validate all addresses in `ValidateBasic()`?

File: x/deployment/types/types.go [hunk 4]
```diff
@@ -17,59 +17,11 @@ const DefaultOrderBiddingDuration = int64(12342)
...
-// 	"active": DeploymentActive,
-// 	"closed": DeploymentClosed,
-// }
-
-// Deployment stores deploymentID, state and version details
-// type Deployment struct {
-// 	DeploymentID `json:"id"`
-// 	State        DeploymentState `json:"state"`
-// 	Version      []byte          `json:"version"`
-// }
```
Comment (boz) [inline x/deployment/types/types.go:43]: :pray:

File: x/market/client/cli/cli_test.go [hunk 5]
```diff
@@ -222,12 +222,12 @@ func (s *IntegrationTestSuite) Test2CreateBid() {
 	resp, err = cli.QueryBidsExec(val.ClientCtx.WithOutputFormat("json"))
 	s.Require().NoError(err)
 
-	var bidRes *types.QueryBidsResponse = &types.QueryBidsResponse{}
+	bidRes := &types.QueryBidsResponse{}
```
Comment (boz) [inline x/market/client/cli/cli_test.go:225]: :pray:

**Review by boz — COMMENTED** [review]
Thanks @akhilkumarpilli .  Some comments inline.  Can you rebase with master so we can queue this up to merge?

File: sdkutil/object_id.go (hunk 2 above)
Comment (boz) [reply sdkutil/object_id.go:110]: @akhilkumarpilli I have replaced these with a simpler method in master.

File: x/deployment/types/msgs.go (hunk 3 above)
Comment (troian) [reply x/deployment/types/msgs.go:40]: @boz @akhilkumarpilli, as we doing `msg.ID.Validate()` (which checks errors from sdk.AccAddressFromBech32) in all `ValidateBasic` can remove panic then

**Comment by troian** [comment]
@akhilkumarpilli there are a bunch of conflicts unfortunately 😿

**Review by troian — APPROVED** [review]

**Review by boz — APPROVED** [review]
:+1:

File: x/deployment/keeper/key.go [hunk 6]
```diff
@@ -17,7 +17,7 @@ var (
 
 func deploymentKey(id types.DeploymentID) []byte {
 	buf := bytes.NewBuffer(deploymentPrefix)
-	buf.Write(id.Owner.Bytes())
+	buf.Write([]byte(id.Owner))
```
Comment (boz) [inline x/deployment/keeper/key.go:20]: This really increases the size of our indexes.

We should look at using `AccAddress` instead.  It's not necessary for this PR tho.

**Review by boz — APPROVED** [review]
Looks good @akhilkumarpilli .  I imagine that the proto check failure is expected?

I do wonder - why did we change all of the fields in IDs to strings tho?

**Comment by boz** [comment]
I'd like to understand the motivation for this before merging it.  There are many benefits to having typed values that we lose when everything is turned into a string.

**Review by boz — APPROVED** [review]
Spoke with @akhilkumarpilli  - this is needed so that json requests to grpc-gateway can use bech strings instead of bytes.

Can look into another way of doing it later.

**Pull request merged**